diff --git a/drivers/i2c/chips/tpa2051d3.c b/drivers/i2c/chips/tpa2051d3.c
index aac05c5..42498fb 100644
--- a/drivers/i2c/chips/tpa2051d3.c
+++ b/drivers/i2c/chips/tpa2051d3.c
@@ -398,7 +398,9 @@ static long tpa2051d3_ioctl(struct file *file, unsigned int cmd,
 			pr_err("unsupported tpa2051 mode %d\n", modeid);
 			return -EINVAL;
 		}
+	#ifndef CONFIG_SOUND_CONTROL_HAX_GPL
 		rc = update_amp_parameter(modeid);
+	#endif
 		pr_info("set tpa2051 mode to %d\n", modeid);
 		break;
 	case TPA2051_SET_PARAM:
